gn3/api/wiki.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+from typing import Any, Dict, List
+from flask import Blueprint, request, jsonify, current_app
+from gn3 import db_utils
+
+
+wiki = Blueprint("wiki", __name__)
+
+
+class MissingDBDataException(Exception):
+    pass
+
+
+@wiki.route("/<int:comment_id>/edit", methods=["POST"])
+def edit_wiki(comment_id: int):
+    # FIXME: attempt to check and fix for types here with relevant errors
+    payload: Dict[str, Any] = request.json
+    pubmed_ids = [str(x) for x in payload.get("pubmed_ids", [])]
+
+    insert_dict = {
+        "Id": comment_id,
+        "symbol": payload["symbol"],
+        "PubMed_ID": " ".join(pubmed_ids),
+        "comment": payload["comment"],
+        "email": payload["email"],
+        "createtime": datetime.datetime.now(datetime.timezone.utc).strftime(
+            "%Y-%m-%d %H:%M"
+        ),
+        "user_ip": request.environ.get("HTTP_X_REAL_IP", request.remote_addr),
+        "weburl": payload.get("web_url"),
+        "initial": payload.get("initial"),
+        "reason": payload["reason"],
+    }
+
+    insert_query = """
+    INSERT INTO GeneRIF (Id, versionId, symbol, PubMed_ID, SpeciesID, comment,
+                         email, createtime, user_ip, weburl, initial, reason)
+    VALUES (%(Id)s, %(versionId)s, %(symbol)s, %(PubMed_ID)s, %(SpeciesID)s, %(comment)s, %(email)s, %(createtime)s, %(user_ip)s, %(weburl)s, %(initial)s, %(reason)s)
+    """
+    with db_utils.database_connection(current_app.config["SQL_URI"]) as conn:
+        cursor = conn.cursor()
+        try:
+            category_ids = get_categories_ids(cursor, payload["categories"])
+            species_id = get_species_id(cursor, payload["species"])
+            next_version = get_next_comment_version(cursor, comment_id)
```

Comment:
*suggestion:* @jnduli I would suggest moving the implementation of `get_categories_ids`, `get_species_id`, and `get_next_comment_version` to their own module.  Thtese functions seem re-usable and would make sense to move them to the db module for more re-use.

Reply:
thanks. I've moved these to `gn3.db.wiki` and added an extra module `gn3.api.metadata_apis.wiki` to house the wiki apis.